gomint-server/src/main/java/io/gomint/server/entity/Entity.java

```diff
@@ -267,11 +268,15 @@ public EntityType type() {
 
     @Override
     public WorldAdapter world() {
-        return this.world;
+        synchronized (this.worldSyncObject) {
+            return this.world;
+        }
     }
 
     public void world(WorldAdapter world ) {
-        this.world = world;
+        synchronized (this.worldSyncObject) {
```

Comment:
This should be removed. Setting a world should be forced to only work on the thread of the world

Reply:
On the thread of the old world or new world?

Reply:
the new one

Reply:
changed. i added a volatile keyword on he world field though, I think that is needed at least. But I am no master in concurrency

Reply:
a entity isn't shared between multiple threads is it? so it doesn't need any protections from my POV